### ISSUE TITLE
Fails to create rds.aws.upbound.io SecurityGroup

### ISSUE THREAD

**Issue by junglie85** [upbound/provider-aws#689]
### What happened?

RDS instance does not deploy because SecurityGroup could not be created:

```
kubectl describe securitygroup.rds.aws.upbound.io/backstage-dbinstance-internal
Name:         backstage-dbinstance-internal
Namespace:
Labels:       backstage.web.com/name=backstage-dbinstance
Annotations:  crossplane.io/external-name: backstage-dbinstance-internal
              web.com/id: rds/v1beta1/securitygroup
API Version:  rds.aws.upbound.io/v1beta1
Kind:         SecurityGroup
Metadata:
  Creation Timestamp:  2023-05-04T09:36:14Z
  Generation:          2
  Resource Version:    2174188
  UID:                 22633834-f58a-44ce-9188-a9541cf3b3b3
Spec:
  Deletion Policy:  Delete
  For Provider:
    Ingress:
      Cidr:  10.0.0.0/16
    Region:  eu-west-1
    Tags:
      Crossplane - Kind:            securitygroup.rds.aws.upbound.io
      Crossplane - Name:            backstage-dbinstance-internal
      Crossplane - Providerconfig:  aws-provider
  Management Policy:                FullControl
  Provider Config Ref:
    Name:  aws-provider
Status:
  At Provider:
  Conditions:
    Last Transition Time:  2023-05-04T09:54:37Z
    Message:               observe failed: cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
                           status code: 404, request id: fb5f3cb3-09f4-4778-b541-6f699cd6761c:
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
Events:
  Type     Reason                         Age   From                                                    Message
  ----     ------                         ----  ----                                                    -------
  Warning  CannotObserveExternalResource  18m   managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: ff6a5c36-74ba-49ee-a50d-59e15425cbec:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 94c629d3-0d52-47b8-ad63-a1fd4943766a:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 9e435978-a9a8-4fc1-8817-13c01c71ba9f:
  Warning  CannotInitializeManagedResource  17m (x2 over 17m)  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  Operation cannot be fulfilled on securitygroups.rds.aws.upbound.io "backstage-dbinstance-internal": the object has been modified; please apply your changes to the latest version and try again
  Warning  CannotObserveExternalResource    17m                managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: dd04f122-65d3-4b4e-8ef0-f12d8fdd4268:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 63805613-1937-49c7-b8a4-b76936b827b1:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 7c504a93-2d17-4ded-8972-427694645a0e:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 998d0394-e847-42d4-98d7-1594a4062a76:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 32a0d34c-9db5-4349-aac1-9dcc0dcb896f:
  Warning  CannotObserveExternalResource  17m  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: 89f0e0da-c21f-4f50-b078-a7fe87de0bb7:
  Warning  CannotObserveExternalResource  3m6s (x371 over 17m)  managed/rds.aws.upbound.io/v1beta1, kind=securitygroup  (combined from similar events): cannot run refresh: refresh failed: reading RDS DB Security Group (backstage-dbinstance-internal): Error retrieving DB Security Groups: DBSecurityGroupNotFound: DBSecurityGroup backstage-dbinstance-internal not found.
           status code: 404, request id: a4de8f5c-5985-48c4-836a-f299a9744666:
```

Logs for the `provider-aws` pod do not show any errors:

```
 kubectl -n crossplane-system logs deployment.apps/provider-aws-6acd9324f315
I0504 08:12:24.402294       1 request.go:690] Waited for 1.004813878s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/apis/connect.aws.upbound.io/v1beta1?timeout=32s
{"level":"info","ts":"2023-05-04T08:12:31Z","logger":"provider-aws","msg":"Alpha feature enabled","flag":"EnableAlphaManagementPolicies"}
I0504 08:12:41.659104       1 trace.go:219] Trace[479797334]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.26.3/tools/cache/reflector.go:169 (04-May-2023 08:12:31.636) (total time: 10022ms):
Trace[479797334]: ---"Objects listed" error:<nil> 10022ms (08:12:41.659)
Trace[479797334]: [10.022638814s] [10.022638814s] END
...
I0504 08:12:46.668927       1 trace.go:219] Trace[1118034089]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.26.3/tools/cache/reflector.go:169 (04-May-2023 08:12:31.878) (total time: 14790ms):
Trace[1118034089]: ---"Objects listed" error:<nil> 14790ms (08:12:46.668)
Trace[1118034089]: [14.790877157s] [14.790877157s] END
I0504 08:12:46.685073       1 trace.go:219] Trace[524228719]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.26.3/tools/cache/reflector.go:169 (04-May-2023 08:12:31.884) (total time: 14800ms):
Trace[524228719]: ---"Objects listed" error:<nil> 14800ms (08:12:46.685)
Trace[524228719]: [14.80044571s] [14.80044571s] END
```

How can I find out what the problem is?

### How can we reproduce it?

```
apiVersion: rds.aws.upbound.io/v1beta1
kind: SecurityGroup
metadata:
  annotations:
    web.com/id: rds/v1beta1/securitygroup
  labels:
    backstage.web.com/name: backstage-dbinstance
  name: backstage-dbinstance-internal
  namespace: backstage
spec:
  providerConfigRef:
    name: aws-provider
  forProvider:
    ingress:
      - cidr: 10.0.0.0/16
    region: eu-west-1
```

### What environment did it happen in?

* Crossplane Version: 1.12.1
* Provider Version: 0.34.0
* Kubernetes Version: 1.26
* Kubernetes Distribution: EKS

**Comment by turkenf** [comment]
Hi @junglie85,

Thank you for raising this issue. I can reproduce it but as @haarchri mentioned in this [comment](https://github.com/upbound/provider-aws/issues/691#issuecomment-1535085056), I think this resource has been removed from AWS, [EC2-Classic was retired on August 15, 2022](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBSecurityGroups.html#:~:text=DB%20security%20group.-,Note,-EC2%2DClassic%20was).

You can use `ec2.SecurityGroup` instead of `rds.SecurityGroup`

**Comment by junglie85** [comment]
I managed to get to that conclusion after making some other changes and was able to get the error message that caused me to raise https://github.com/upbound/provider-aws/issues/691

I think the issue here was assuming that I could create RDS security groups by name, which was failing. When I added the `external-name/default` annotation the error was clearer.

I think it would be helpful to remove the deprecated API elements and explain the deprecation in the docs.
